### ISSUE TITLE
Zig-zag detections in inference

### ISSUE THREAD

**Issue by AlbertoPad** [ultralytics/ultralytics#24978]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I'd like to know why, in version 11 of YOLO, when I run an inference on a model (yolo11m-seg), I get those lines that try to connect various detections.

<img width="604" height="334" alt="Image" src="https://github.com/user-attachments/assets/65dc996c-9e26-4db5-a0b8-caea5fb7ce8b" />

### Environment

Name: ultralytics
Version: 8.4.6
Summary: Ultralytics YOLO 🚀 for SOTA object detection, multi-object tracking, instance segmentation, pose estimation and image classification.
Home-page: https://ultralytics.com/
Author: 
Author-email: Glenn Jocher [glenn.jocher@ultralytics.com](mailto:glenn.jocher@ultralytics.com), Jing Qiu [jing.qiu@ultralytics.com](mailto:jing.qiu@ultralytics.com)
License: AGPL-3.0
Location: /home/asimob/Desktop/asimobEnv/lib/python3.11/site-packages
Requires: matplotlib, numpy, opencv-python, pillow, polars, psutil, pyyaml, requests, scipy, torch, torchvision, ultralytics-thop
Required-by:

Model Hyperparameters
{
  "lr0": 0.01,
  "lrf": 0.001,
  "momentum": 0.937,
  "weight_decay": 0.0005,
  "warmup_epochs": 3.0,
  "warmup_momentum": 0.8,
  "warmup_bias_lr": 0.1,
  "box": 7.5,
  "cls": 0.5,
  "dfl": 1.5,
  "pose": 0.0,
  "kobj": 1.0,
  "label_smoothing": 0.1,
  "nbs": 64,
  "hsv_h": 0.01,
  "hsv_s": 0.3,
  "hsv_v": 0.3,
  "degrees": 5.0,
  "translate": 0.05,
  "scale": 0.3,
  "shear": 0.0,
  "perspective": 0.0,
  "flipud": 0.0,
  "fliplr": 0.5,
  "mosaic": 1.0,
  "close_mosaic": 25,
  "mixup": 0.15,
  "copy_paste": 0.3,
  "erasing": 0.4,
  "imgsz": 1280,
  "overlap_mask": true,
  "retina_masks": true,
  "mask_ratio": 2,
  "rect": false
}

### Minimal Reproducible Example

        # --- Build and log training parameters ---
        trainingParameters = {"data": str(pathToYOLOYaml),"epochs": epochs,"imgsz": imgsz,"batch": batch,"save_period": 1,
                              "patience": patience,"project": str(savePath),"name": "train","exist_ok": True,"resume": checkPoint, **dataJsonDict}
        logger.info("Training parameters:\n%s",json.dumps({k: str(v) for k, v in trainingParameters.items()}, indent=2))
        # --- Train ---
        model = YOLO(pretrainedModel)
        model.train(**trainingParameters)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AlbertoPad, thank you for reporting this and for sharing the training details and screenshot 🚀 This is an automated response to help gather the right debugging information, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.

Since this appears to be a 🐛 Bug Report, please provide a complete [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) to help us debug it. In particular, please include:
- The exact inference code used to generate the zig-zag lines
- Whether the lines appear in saved plots, returned masks, or only in visualization
- A small test image and, if possible, the corresponding model output
- The exact command or script used for inference
- Confirmation that the issue persists after upgrading to the latest package version

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Please upgrade to the latest `ultralytics` package, since your current version is `8.4.6`, and verify whether the behavior is still reproducible ✅ Upgrade including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Hello.

This is just an artefact of trying to represent multiple masks with a single polygon which can't be disconnected. For segmentation, it's recommended to use the masks (`result.masks.data`) instead of polygons.

https://docs.ultralytics.com/tasks/segment#predict
